### PR TITLE
Add documentation for `MergeSources`

### PR DESCRIPTION
**Pull request by londospark**
- I haven't filled in the Expression -> Translation table starting at line 286 as I don't know the exact translation.
- Not 100% sure that the added documentation is good, but wanted to have them in as a starting point for adding the Applicative CE feature to this page.

## Summary

Started to add documentation for `MergeSources` to the CE page in the language reference

### CODE REVIEW

**Comment by BillWagner** [comment]
adding @dsyme for review.

File: docs/fsharp/language-reference/computation-expressions.md [hunk 1]
```diff
@@ -77,6 +78,29 @@ If you bind the call to a computation expression with `let`, you will not get th
...
+### `and!`
+
+The `and!` keyword binds the result of a call to another computation expression to a name, unlike `let!` it does not allow you to use previous results:
+
+```fsharp
+// OK
+let doThingsAsync url otherUrl =
+    async {
+        let! data = getDataAsync url
+        and! moreData = getDataAsync otherURL
```
Comment (dsyme) [inline docs/fsharp/language-reference/computation-expressions.md:90]: The problem here is that, by default, `async` doesn't have support for `and!`, and no built-in computation expression does.

I'm not sure what to do about this.  Perhaps just adjust the example to use one of the computation expressions here: https://github.com/dsyme/fsharp-presentations/blob/master/design-notes/rethinking-applicatives.md.  Then link to that as a resource at the end.  Anything more will require an extensive rewrite of the whole page I think.

File: docs/fsharp/language-reference/computation-expressions.md [hunk 2]
```diff
@@ -231,6 +255,7 @@ The following table describes methods that can be used in a workflow builder cla
 |**Method**|**Typical signature(s)**|**Description**|
 |----|----|----|
 |`Bind`|`M<'T> * ('T -> M<'U>) -> M<'U>`|Called for `let!` and `do!` in computation expressions.|
+|`MergeSources`|`M<'T> * M<'U> -> M<'T * 'U>`|Called for `and!` in computation expressions.|
```
Comment (dsyme) [inline docs/fsharp/language-reference/computation-expressions.md:258]: Please also 

1. Also add `Bind2`, `Bind3`, ... mentioned in the RFC
2. Also add `BindReturn`, `BindReturn2`, `BindReturn3`, ... mentioned in the RFC
3. Also add a link to the RFC https://github.com/fsharp/fslang-design/blob/main/FSharp-5.0/FS-1063-support-letbang-andbang-for-applicative-functors.md
4. Also add a link to  https://github.com/dsyme/fsharp-presentations/blob/master/design-notes/rethinking-applicatives.md

Comment (dsyme) [reply docs/fsharp/language-reference/computation-expressions.md:258]: If this is too much, I can add these, let us know :)

File: docs/fsharp/language-reference/computation-expressions.md [hunk 3]
```diff
@@ -77,6 +78,29 @@ If you bind the call to a computation expression with `let`, you will not get th
...
+    
+// Error
+let doThingsAsync url otherUrl =
+    async {
+        let! data = getDataAsync url
+        and! moreData = processDataAsync data // Error 'data' is not defined
+        ...
+    }
+```
+If you use `and!` then each computation expression may be bound in parallel, therefore the bindings may not depend on other bindings. All of the values will be retrieved and then collated at the end of the computation expression.
```
Comment (IEvangelist) [inline docs/fsharp/language-reference/computation-expressions.md:102]:
```suggestion

If you use `and!` then each computation expression may be bound in parallel, therefore the bindings may not depend on other bindings. All of the values will be retrieved and then collated at the end of the computation expression.
```

**Comment by BillWagner** [comment]
@londospark 

I'm closing this as very stale. It's been more than 2 years since the last comment, and there are now conflicts. If you want to reopen the PR, address the conflicts and the comments, we'll take another look.

/cc @dsyme

**Pull request closed without merging**